code/src/stf/include/interface/stf/alg/clipping.hpp

```diff
@@ -0,0 +1,161 @@
+#pragma once
+
+#include "stf/geom/segment.hpp"
+#include "stf/geom/polyline.hpp"
+
+/**
+ * @file clipping.hpp
+ * @brief A file containing functions that clip geometric objects
+ */
+
+/// @cond DELETED
+namespace stf::alg::guts
```

Comment:
An internal namespace meant for use in this file. Not intended to be part of the public API